src/lib.rs

```diff
@@ -837,3 +936,61 @@ impl Command {
         async { child?.output().await }
     }
 }
+
+#[cfg(unix)]
+/// Moves `Fd` out of nonblocking mode.
+pub fn set_blocking<T: std::os::unix::io::AsRawFd>(stdout: T) -> io::Result<T> {
```

Comment:
When adopting the `into_inner` + `set_blocking` approach, I would prefer to replace this bounds with a [sealed trait](https://rust-lang.github.io/api-guidelines/future-proofing.html#sealed-traits-protect-against-downstream-implementations-c-sealed).

Reply:
```suggestion
/// Moves `Fd` out of nonblocking mode.
pub fn set_blocking<T: Blocking>(stdout: T) -> io::Result<T> {
    unsafe {
        let res = libc::fcntl(stdout.as_raw_fd(), libc::F_GETFL);
        let errno = libc::fcntl(
            stdout.as_raw_fd(),
            libc::F_SETFL,
            !(!res | libc::O_NONBLOCK),
        );

        // Unix-like systems when errno is_minus_one then return last_os_error.
        if errno == -1 {
            return Err(io::Error::last_os_error());
        }
    }

    Ok(stdout)
}

/// This is a sealed trait that allows types with the specified range of constraints to use `set_bloking`.
pub trait Blocking: secret_jar::SuperBlocking {}

/// Implement Blocking to `ChildStd*`, because getting the internal `ChildStd*` via `into_inner`
/// may require restoring the file descriptor to blocking mode.
impl Blocking for ChildStdin {}
impl Blocking for ChildStdout {}
impl Blocking for ChildStderr {}

mod secret_jar {
    use std::os::unix::io::AsRawFd;
    use std::process::{ChildStderr, ChildStdin, ChildStdout};

    pub trait SuperBlocking: AsRawFd {}

    impl SuperBlocking for ChildStdin {}
    impl SuperBlocking for ChildStdout {}
    impl SuperBlocking for ChildStderr {}
}
```



I think your suggestion makes sense **thank you very much**, we can use seal-trait to put the control in our hands. I added the following code to unix-mod, what do you think of this?

Reply:
Hi, I'm sorry to bother you, I updated my pr hope you have time to review it, if there are unreasonable places I promptly correct, very much hope to support this feature. @taiki-e

Reply:
Hi @taiki-e , I'm sorry to bother you, I updated my pr hope you have time to review it, if there are unreasonable places I promptly correct, very much hope to support this feature.